detox/android/detox/src/main/java/com/wix/detox/espresso/scroll/SwipeHelper.kt

```diff
@@ -0,0 +1,114 @@
+package com.wix.detox.espresso.scroll
+
+import android.view.View
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.*
+import com.wix.detox.common.DetoxErrors
+import com.wix.detox.espresso.common.annot.*
+import com.wix.detox.espresso.utils.AgnosticPoint2D
+import kotlin.math.min
+import kotlin.math.max
+
+private const val EDGE_FUZZ_FACTOR = 0.083f
+
+typealias CreateSwipeAction = (
+        swiper: Swiper,
+        startCoordinatesProvider: CoordinatesProvider,
+        endCoordinatesProvider: CoordinatesProvider,
+        precisionDescriber: PrecisionDescriber
+) -> ViewAction
+
+class SwipeHelper(private val createAction: CreateSwipeAction) {
+
+    fun swipeInDirection(
+            direction: Int,
+            fast: Boolean = true,
+            offset: Double = Double.NaN,
+            startPositionX: Double = Double.NaN,
+            startPositionY: Double = Double.NaN
+    ): ViewAction {
+        val unsafeStartingPoint = AgnosticPoint2D.fromDoubles(startPositionX, startPositionY, direction)
+        val startingPoint = getSafeStartingPoint(unsafeStartingPoint, direction)
+        val start = translateFrom(startingPoint, direction)
+        val end = translateTo(startingPoint, direction, offset)
```

Comment:
Naming here also had me a bit confused, because `translate` in the context of points really makes you think you're getting a translated point. Could something like `getFromCoordsProvider()` work?

Reply:
Sure, I'll align the naming to iOS tomorrow.